_drafts/gdb-python-post.md

```diff
@@ -0,0 +1,378 @@
+
+       print(self._uuid_list_to_str(node_ptr_val))
+```
+
+To actually load the command all we need to do is instantiate the object by adding:
+
+```python
+UuidListDumpCmd()
+```
+
+The command should then appear in our user-defined list:
+
+```
+(gdb) help user-defined
+User-defined commands.
+The commands in this class are those defined by the user.
+Use the "define" command to define a command.
+
+List of commands:
+
+uuid_list_dump -- Prints the ListNode from our example in a nice format!
+```
+
+Let's try it out!
+
+```
+(gdb) uuid_list_dump s_list_head
+Args Passed: s_list_head
+Found a Linked List with 10 nodes:
+0: Addr: 0x200070c0, uuid: 9d5d4440-8b5a-235e-a9ef-ec3f919c7a41
+1: Addr: 0x200070a8, uuid: 8afe211b-197e-e475-8ec6-878558dd24e6
+2: Addr: 0x20007088, uuid: 0f84cd49-d683-4217-e7a3-399373c6e6f1
+3: Addr: 0x20007070, uuid: 23b05b1b-a32a-8288-4419-0e30f26ba3ae
+4: Addr: 0x20007050, uuid: 3ca835e5-8f2a-cdc2-40c1-cf52716a7229
+5: Addr: 0x20007038, uuid: 30d4349d-3a0d-0fbd-2fa1-f70fd968f4d9
+6: Addr: 0x20007018, uuid: 437bbe90-1689-9d7e-77c6-2f269888f5b4
+7: Addr: 0x20007000, uuid: 5e20fb38-a84e-a614-9325-562444df598d
+8: Addr: 0x20006fe0, uuid: 7a96092c-a557-7464-c4af-1528a4e957db
+9: Addr: 0x20006fc8, uuid: 2dcf4629-04b4-78d8-68a7-ff3f2bf1fcd9
+```
+
+Or we could even start dumping the list in the middle. For example, let's take the Address at entry `5`:
+
+```
+(gdb) uuid_list_dump (UuidListNode*)0x20007038
+Args Passed: (UuidListNode*)0x20007038
+Found a Linked List with 5 nodes:
+0: Addr: 0x20007038, uuid: 30d4349d-3a0d-0fbd-2fa1-f70fd968f4d9
+1: Addr: 0x20007018, uuid: 437bbe90-1689-9d7e-77c6-2f269888f5b4
+2: Addr: 0x20007000, uuid: 5e20fb38-a84e-a614-9325-562444df598d
+3: Addr: 0x20006fe0, uuid: 7a96092c-a557-7464-c4af-1528a4e957db
+4: Addr: 0x20006fc8, uuid: 2dcf4629-04b4-78d8-68a7-ff3f2bf1fcd9
+```
+
+### Auto Loading scripts
+
+You can automatically load custom extensions by adding the following to your [gdbinit](https://sourceware.org/gdb/onlinedocs/gdb/gdbinit-man.html) file which is invoked anytime gdb starts
+
+```
+source /path/to/custom_gdb_extensions.py
+```
+
+Alternatively, you could also get this to run automatically when gdb is started by adding `--ex="source /path/to/custom_gdb_extensions.py"` as an argument
+
+## Closing
+
+We hope this post gave you a useful overview of how to add custom GDB commands and pretty printers using the Python API. Sometimes adding GDB commands like these can even be used to reduce [code size](https://interrupt.memfault.com/blog/best-firmware-size-tools) by replacing a on-device CLI command that would display the same information!
+
+Do you already have some ideas about how the python API could be applied to automate parts of your debugging flow ... perhaps a command to walk custom heaps or display the contents in a memory-mapped filesystem? Or maybe you are already have some great examples of how you have used GDB Python? Either way, let us know in the discussion area below!
```

Comment:
Make sure to add a link to an example repo with the source code and the exactly Python script they could source.

Reply:
You could add the source files to the `interrupt` repo since it's open source already

Reply:
Works for me! I was going to tack it into zero-to-main but I thing it probably does just make more sense to have an example/ directory in this repo